File: day17/src/main/java/de/havox_design/aoc2016/day17/Direction.java
package de.havox_design.aoc2016.day17;

import java.util.Arrays;

public enum Direction {
    UP('U'),
    DOWN('D'),
    LEFT('L'),
    RIGHT('R');

    private char symbol;

    Direction(char symbol) {
        this.symbol = symbol;
    }

    public char getSymbol() {
        return symbol;
    }

    public static Direction from(char symbol) {
        return Arrays
                .stream(values())
                .filter(dir -> dir.symbol == symbol)
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Unknown symbol '" + symbol + "'."));
    }
}

File: day08/src/main/kotlin/de/havox_design/aoc2023/day08/Direction.kt
package de.havox_design.aoc2023.day08

enum class Direction(private val symbol: Char) {
    LEFT('L'),
    RIGHT('R');

    companion object {
        fun from(char: Char): Direction {
            if(entries.map { dir -> dir.symbol }.contains(char)) {
                return entries.first { dir -> dir.symbol == char }
            }

            throw IllegalArgumentException("Unknown symbol '${char}'.")
        }
    }
}